ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,140 @@
+
+*Remark*.
+If you disagree with the decisions in this document, please consider reviewing our brainstorming notes and discussions, available at [this document's PR](https://github.com/input-output-hk/ouroboros-network/pull/4207).
+Even if you choose not to do so, please share your concern with us.
+
+## Notation
+
+In this document, we will assume each version number is a MAJOR.MINOR.PATCH triple.
+This is easy to generalize to other conventional notations, such as [the PVP](https://pvp.haskell.org/)'s MAJOR.MAJOR.MINOR.PATCH quadruples.
+
+We refer throughout this document to _the main branch_ (aka `master`, aka `main`).
+
+If you search for "RULE:" in this document, you'll find the key statements.
+
+## Rules for Branches
+
+The Consensus team will sometimes be maintaining multiple MAJOR.MINOR versions of the same package simultaneously.
+EG We might release FOO-2.4.1 on Monday, release FOO-2.3.7 on Wednesday, and release FOO-2.4.2 on Friday, etc.
+
+- Whenever we're maintaining only the one greatest-ever MAJOR.MINOR version of a package, then all work on that package will happen on the main branch.
+- Whenever we're also maintaining some lesser MAJOR.MINOR versions of a package, then some work for that package will also be happening on the respective _release branches_.
+
+A _release branch_ is a branch dedicated to the maintenance of some older release of a package.
+
+RULE: If we have released a package FOO-X.Y.Z, then the release-FOO-X.Y.x branch MUST exist and its first-parent history MUST include the commit released as FOO-X.Y.Z.
+
+There are two kinds of work on release branches.
+Most of the time, that work is immitating some work that was done on the main branch.
+EG we fixed a bug on the main branch and we're backporting that fix onto a release branch (hopefully it's a trivial cherry-pick).
+Rarely, though, there will be fresh work done on a release branch.
+EG Consider the following possible timeline.
+
+- We release FOO-2.4.1 on Monday, and FOO-2.4.1 completely reworked some functionality.
+- We then realize on Tuesday that there was a severe bug in the old functionality.
+- We fix the bug by merging a PR that targets the FOO-2.3.x release branch (since the bug no longer exists on the main branch!)
+- We release FOO-2.3.7 on Wednesday.
+
+*Remark*.
+Not every first-parent commit in the history of the release branch is a release commit.
+We will be merging multiple PRs into the release branch in order to prepare the next release from it, so some commits will just be the intermediates between two releases.
+
+*Remark*.
+The release branch for the greatest MAJOR.MINOR version of a package is somewhat degenerate.
+It's either equal to or a prefix of the main branch.
+The only time it MUST be updated to the tip of the main branch is when we cut a release of the package from the main branch, to satisfy the rule above about all X.Y.Z release commits being on the release-X.Y.x branch.
```

Comment:
It'd be nice to have an example of this.

Reply:
Similar to the example above:

```text
            FOO-2.3.6 --- FOO-2.3.7 <- release branch 
             /                                            
-------FOO-2.3.? ------  .... ------ FOO-2.4.1 --- (what happens with the commits here?)  
                                        ^
                                        |
                                   main branch = release branch? 
```                                   

do we need to create a branch when we release `2.4.1`? If so, what happens to the commits on `main`?

Reply:
I pushed up a longer example. Does it suffice for this?

Reply:
Yes.